server/src/test/java/org/bonitasoft/console/common/server/servlet/IconServletTest.java

```diff
@@ -0,0 +1,146 @@
+/**
+ * Copyright (C) 2016 Bonitasoft S.A.
+ * Bonitasoft, 32 rue Gustave Eiffel - 38000 Grenoble
+ * This library is free software; you can redistribute it and/or modify it under the terms
+ * of the GNU Lesser General Public License as published by the Free Software Foundation
+ * version 2.1 of the License.
+ * This library is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY;
+ * without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+ * See the GNU Lesser General Public License for more details.
+ * You should have received a copy of the GNU Lesser General Public License along with this
+ * program; if not, write to the Free Software Foundation, Inc., 51 Franklin Street, Fifth
+ * Floor, Boston, MA 02110-1301, USA.
+ **/
+
+package org.bonitasoft.console.common.server.servlet;
+
+import static org.mockito.Mockito.*;
+
+import javax.servlet.ServletOutputStream;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+import org.bonitasoft.engine.api.IdentityAPI;
+import org.bonitasoft.engine.exception.NotFoundException;
+import org.bonitasoft.engine.identity.impl.IconImpl;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.Spy;
+import org.mockito.runners.MockitoJUnitRunner;
+
+/**
+ * @author Baptiste Mesta
+ */
+@RunWith(MockitoJUnitRunner.class)
+public class IconServletTest {
+
+    private static final long ICON_ID = 1238970432L;
+    private static final String ICON_ID_STRING = "1238970432";
+    private static final byte[] ICON_CONTENT = "content".getBytes();
+    @InjectMocks
+    @Spy
+    private IconServlet iconServlet;
+    @Mock
+    private HttpServletRequest httpServletRequest;
+    @Mock
+    private HttpServletResponse httpServletResponse;
+    @Mock
+    private IdentityAPI identityAPI;
+    @Mock
+    private ServletOutputStream outputStream;
+
+    @Before
+    public void before() throws Exception {
+        doReturn(identityAPI).when(iconServlet).getIdentityApi(httpServletRequest);
+        doReturn(outputStream).when(httpServletResponse).getOutputStream();
+    }
+
+    private void havingIcon(String mimeType, long iconId) throws NotFoundException {
+        doReturn(new IconImpl(iconId, mimeType, ICON_CONTENT)).when(identityAPI).getIcon(iconId);
+    }
+
+    private HttpServletRequest request(String contextUrl) {
+        doReturn(contextUrl).when(httpServletRequest).getPathInfo();
+        return httpServletRequest;
+    }
+
+    @Test
+    public void should_get_return_user_icon_from_engine() throws Exception {
```

Comment:
In my mind it was: should <subject> <do something> <when something>
i.e. should get (=the subjet) return user icon (=the action) [] (= the condition, i.e. normal case)
But I see your point,
is this better: "should return user icon from engine when valid id is given in url"
However in case of class with more than one entry point I'd prefer having the method name inside the test name, Whats your opinion on this? better having it at the start (subject, action, condition), better having it in the condition (e.g. when get is called with valid id in url), or not having it because its enouth to have it in the test?